### ISSUE TITLE
Averaging method utils.federated_avg(models) returns wrong results and further bug

### ISSUE THREAD

**Issue by PatrickITWM** [OpenMined/PySyft#3863]
## Description
The result of the function utils.federated_avg() is not the average of the input models.

## How to Reproduce
class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv1 = nn.Conv2d(1, 10, kernel_size=5)
        self.conv2 = nn.Conv2d(10, 20, kernel_size=5)
        self.conv2_drop = nn.Dropout2d()
        self.fc1 = nn.Linear(320, 50)
        self.fc2 = nn.Linear(50, 10)

    def forward(self, x):
        x = F.relu(F.max_pool2d(self.conv1(x), 2))
        x = F.relu(F.max_pool2d(self.conv2_drop(self.conv2(x)), 2))
        x = x.view(-1, 320)
        x = F.relu(self.fc1(x))
        x = F.dropout(x, training=self.training)
        x = self.fc2(x)
        return F.log_softmax(x,dim=1)

net1 = Net()
net2 = Net()
avg_net = utils.federated_avg(dict(enumerate([net1, net2])))
print("Net 1  ", list(net1.parameters())[0][0][0][0])
print("Net 2  ", list(net2.parameters())[0][0][0][0])
print("----------------------------")
print("fed_avg", list(avg_net.parameters())[0][0][0][0])
print("----------------------------")
print("avg 1+2", 1/2*(list(net1.parameters())[0][0][0][0]+list(net2.parameters())[0][0][0][0]))
print("----------------------------")
print("Net 1  ", list(net1.parameters())[0][0][0][0])
print("Net 2  ", list(net2.parameters())[0][0][0][0])

## Expected Behavior
The result should be the average.
## Screenshots
![FA Not Correct 3](https://user-images.githubusercontent.com/68423474/87773999-3f06a780-c824-11ea-8284-330806811356.png)

## What is the Problem and how to fix?

In the method utils.federated_avg(models) you initially use
**model = type(model_list[0])()**
to sum over the input models. Unfortunatelly, model has **non zero parameters**, which causes the wrong result.
A fix could be 
![Hot Fix FA Error in Initialize](https://user-images.githubusercontent.com/68423474/87774354-b50b0e80-c824-11ea-8dd5-a9eb38c4dc10.png)

**Comment by PatrickITWM** [comment]
There is a further Bug in the averaging Method: If two Nets living on the GPU are averaged, one gets an exception because the new model for the sum is initiated on the CPU.

![average cuda bug](https://user-images.githubusercontent.com/68423474/89156140-7b4b3f00-d56a-11ea-9519-3a3ea362f0ad.png)

**Comment by aanurraj** [comment]
Hi, I would like to work on this bug. Can I?

**Comment by PatrickITWM** [comment]
What is the current state of this issue?
